zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/validator/AuthorizationRequestValidator.java

```diff
@@ -8,34 +8,45 @@
 package io.camunda.zeebe.gateway.rest.validator;
 
 import static io.camunda.zeebe.gateway.rest.validator.ErrorMessages.ERROR_MESSAGE_EMPTY_ATTRIBUTE;
+import static io.camunda.zeebe.gateway.rest.validator.ErrorMessages.ERROR_MESSAGE_EMPTY_NESTED_ATTRIBUTE;
 import static io.camunda.zeebe.gateway.rest.validator.RequestValidator.createProblemDetail;
 
-import io.camunda.zeebe.gateway.protocol.rest.AuthorizationAssignRequest;
+import io.camunda.zeebe.gateway.protocol.rest.AuthorizationPatchRequest;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
 import org.springframework.http.ProblemDetail;
 
 public final class AuthorizationRequestValidator {
   public static Optional<ProblemDetail> validateAuthorizationAssignRequest(
-      final AuthorizationAssignRequest authorizationAssignRequest) {
+      final AuthorizationPatchRequest authorizationPatchRequest) {
     final List<String> violations = new ArrayList<>();
-    if (authorizationAssignRequest.getOwnerKey() == null) {
-      violations.add(ERROR_MESSAGE_EMPTY_ATTRIBUTE.formatted("ownerKey"));
+    if (authorizationPatchRequest.getAction() == null) {
+      violations.add(ERROR_MESSAGE_EMPTY_ATTRIBUTE.formatted("action"));
     }
 
-    if (authorizationAssignRequest.getOwnerType() == null) {
-      violations.add(ERROR_MESSAGE_EMPTY_ATTRIBUTE.formatted("ownerType"));
+    if (authorizationPatchRequest.getResourceType() == null) {
+      violations.add(ERROR_MESSAGE_EMPTY_ATTRIBUTE.formatted("resourceType"));
     }
 
-    if (authorizationAssignRequest.getResourceKey() == null
-        || authorizationAssignRequest.getResourceKey().isBlank()) {
-      violations.add(ERROR_MESSAGE_EMPTY_ATTRIBUTE.formatted("resourceKey"));
-    }
+    if (authorizationPatchRequest.getPermissions() == null
+        || authorizationPatchRequest.getPermissions().isEmpty()) {
+      violations.add(ERROR_MESSAGE_EMPTY_ATTRIBUTE.formatted("permissions"));
+    } else {
 
-    if (authorizationAssignRequest.getResourceType() == null
-        || authorizationAssignRequest.getResourceType().isBlank()) {
-      violations.add(ERROR_MESSAGE_EMPTY_ATTRIBUTE.formatted("resourceType"));
+      authorizationPatchRequest
+          .getPermissions()
+          .forEach(
+              permission -> {
+                if (permission.getPermissionType() == null) {
+                  violations.add(ERROR_MESSAGE_EMPTY_ATTRIBUTE.formatted("permissionType"));
+                } else if (permission.getResourceIds() == null
+                    || permission.getResourceIds().isEmpty()) {
+                  violations.add(
+                      ERROR_MESSAGE_EMPTY_NESTED_ATTRIBUTE.formatted(
+                          "resourceIds", permission.getPermissionType()));
+                }
+              });
```

Comment:
🔧  There is quite some nesting here, some is avoidable of course but I wonder if we could introduce some helper methods to make this block easier to understand, for example a method such as `validateResourceIds` that lifts the else if block out.